store/aciinfo.go

```diff
@@ -21,46 +21,57 @@ import (
 	"time"
 )
 
+// ACIFetchInfo is used to store information about the fetching process of an ACI.
+type ACIFetchInfo struct {
```

Comment:
I didn't understand what you meant. Are you saying that it's not good because it accomplishes the two items you listed instead of having a single responsibility? if so those items are very similar.
I am not happy about passing `ACIFetchInfo` to `WriteACI` so that it can pass it to the constructor. But, a better solution would require a bigger refactoring.
I am up for suggestions, anything but passing the fields of `ACIFetchInfo` as arguments though.